nse_main.lua

```diff
@@ -416,7 +416,9 @@ do
       timeouts[self.host][self.co] = true;
     end
     -- storing script's start time so as to account for script's timeout later
-    self.start_time = os.time()
+    if cnse.script_timeout then
+      self.start_time = os.time()
```

Comment:
I'd just leave this as it was before: it could be useful to have this info available for other features other than --script-timeout